flink/v1.17/flink/src/test/java/org/apache/iceberg/flink/source/TestIcebergSourceWithWatermarkExtractor.java

```diff
@@ -0,0 +1,375 @@
+
+    assertThat(
+            CompletableFuture.supplyAsync(
+                () -> {
+                  int count = 0;
+                  while (count < expectedRecords.size() && iterator.hasNext()) {
+                    received.add(iterator.next());
+                    count++;
+                  }
+                  if (count < expectedRecords.size()) {
+                    throw new IllegalStateException(
+                        String.format("Fail to get %d records.", expectedRecords.size()));
+                  }
+                  return true;
+                }))
+        .succeedsWithin(DEFAULT_COLLECT_DATA_TIMEOUT);
+
+    Set<RowData> expected =
+        expectedRecords.stream()
+            .map(e -> RowDataConverter.convert(TestFixtures.TS_SCHEMA, e))
+            .collect(Collectors.toSet());
+    Assert.assertEquals(expected, received);
+  }
+
+  protected void waitForRecords(CollectResultIterator<RowData> iterator, int num) {
+    assertThat(
+            CompletableFuture.supplyAsync(
+                () -> {
+                  int count = 0;
+                  while (count < num && iterator.hasNext()) {
+                    iterator.next();
+                    count++;
+                  }
+                  if (count < num) {
+                    throw new IllegalStateException(String.format("Fail to get %d records.", num));
+                  }
+                  return true;
+                }))
+        .succeedsWithin(DEFAULT_COLLECT_DATA_TIMEOUT);
+  }
+
+  private CollectResultIterator<RowData> addCollectSink(DataStream<RowData> stream) {
+    TypeSerializer<RowData> serializer =
+        stream.getType().createSerializer(stream.getExecutionConfig());
+    String accumulatorName = "dataStreamCollect_" + UUID.randomUUID();
+    CollectSinkOperatorFactory<RowData> factory =
+        new CollectSinkOperatorFactory<>(serializer, accumulatorName);
+    CollectSinkOperator<RowData> operator = (CollectSinkOperator<RowData>) factory.getOperator();
+    CollectStreamSink<RowData> sink = new CollectStreamSink<>(stream, factory);
+    sink.name("Data stream collect sink");
+    stream.getExecutionEnvironment().addOperator(sink.getTransformation());
+    return new CollectResultIterator<>(
+        operator.getOperatorIdFuture(),
+        serializer,
+        accumulatorName,
+        stream.getExecutionEnvironment().getCheckpointConfig());
+  }
+
+  private Optional<Gauge<Long>> findAlignmentDriftMetric(JobID jobID, long withValue) {
+    String metricsName = SOURCE_NAME + ".*" + MetricNames.WATERMARK_ALIGNMENT_DRIFT;
+    return reporter.findMetrics(jobID, metricsName).values().stream()
+        .map(m -> (Gauge<Long>) m)
+        .filter(m -> m.getValue() == withValue)
+        .findFirst();
+  }
+
+  private GenericAppenderHelper appender() {
+    // We need to create multiple splits, so we need to generate parquet files with multiple offsets
```

Comment:
I wanted to have a test case where we have multiple splits.
This way we can be sure that we do not need to prevent splitting big files. Related to [this](https://github.com/apache/iceberg/pull/8553#discussion_r1394617942) comment.

Reply:
oh. I missed that we set the `splitSize` to 100 in the source builder